cpp/test/Ice/objects/Test.ice

```diff
@@ -90,7 +90,13 @@ sequence<Base> BaseSeq;
 
 class CompactExt;
 
-class Compact(1)
+enum CompactIdEnum { First = 1 }
+
+class Compact(First)
+{
+}
+
+class CompactScoped(CompactIdEnum::First)
```

Comment:
This PR also adds a test for specifying a class's compact-ID using an enumerator.
I agree, this seems like a super crazy thing to do, but we currently support it, and I can't see a single test for this functionality. Whether we should remove this feature is a separate question. If we support something, we should test it.